File: Rule.java
import java.util.Scanner;

public class DivisibleBy16 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a number: ");
        int num = sc.nextInt();

        int lastFour = num % 10000;

        if (lastFour % 16 == 0)
            System.out.println(num + " is divisible by 16");
        else
            System.out.println(num + " is not divisible by 16");
    }
}

File: Rule.cpp
#include <iostream>
using namespace std;

int main() {
    int num;

    cout << "Enter a number: ";
    cin >> num;

    int lastFour = num % 10000;

    if (lastFour % 16 == 0)
        cout << num << " is divisible by 16";
    else
        cout << num << " is not divisible by 16";

    return 0;
}
